### PR TITLE
Release 0.7.61 — physical-copy management from the book summary

### PR DESCRIPTION
**Pull request by fabiodalez-dev**
Release candidate for **0.7.61-rc.1**.

## What's in it
Physical-copy management from the book summary page, with the book/copies and circulation lifecycle made atomic and derived from the copies:

- **Copie Fisiche** section on `/admin/books/{id}` for every book (add-copy modal, per-copy status editing, per-copy delete). Out-of-circulation copies (lost/damaged/maintenance/restoration/transfer) lower the derived total on their own.
- Zero-copy books are valid; on the edit form the copy count is read-only and delegates to per-copy management.
- Book creation and the bulk `increase-copies` endpoint are transactional and allocate collision-free inventory codes; adding an available copy repairs blocked reservations and promotes the wait-list, mirroring the loan engine.
- Loan/reservation hardening: legacy reservations with a missing/past start date are no longer promoted into back-dated loans; `prestato`/`prenotato` stay owned by the loan system.
- Admin copy routes are fixed English literals, inventory-code allocation escapes LIKE metacharacters, copy notes are sanitised/length-capped, and all new strings are translated across the five locales.

## Release notes
- Version: `0.7.61-rc.1` (`version.json` + CHANGELOG updated).
- **No schema change — no migration.**

## Verification (local)
- CI quality mirror (`scripts/ci-quality-local.sh`): PHPStan level 5, composer/npm audits, locale parity, frontend lint, soft-delete guard, migration-version guard, standalone unit tests (no-skip), schema/migration behavioural gate — all green.
- E2E: copy-management (27), atomic-book-create (17 unit + 9 E2E), loan-edge-cases (64), loan-reservation (21), extra-features (20), swal-books (18), mobile-api (83), book-fields-form (4), full-test (137 passed / 8 skipped / 0 failed).

## Summary by CodeRabbit

* **Nuove funzionalità**
  * Gestione completa delle copie fisiche: aggiunta, modifica, eliminazione, stati dedicati e stampa delle etichette.
  * Disponibilità e contatori aggiornati automaticamente in base alle copie.
  * Creazione di libri senza copie e generazione sicura dei codici inventario.
  * La lingua dell’applicazione viene assegnata automaticamente ai nuovi account e può essere personalizzata.

* **Correzioni**
  * Migliorati prestiti, prenotazioni, liste d’attesa e promozioni automatiche.
  * Rafforzate validazioni, transazioni e gestione delle date legacy.
  * Aggiornate le traduzioni in cinque lingue.

* **Test**
  * Ampliate le verifiche end-to-end, di accessibilità e dei casi limite.

### CODE REVIEW

File: tests/issue-173-cover-save.spec.js [hunk 1]
```diff
@@ -87,8 +87,18 @@ test.describe('#173 — external covers download & save as a local file', () =>
...
-    const id = parseInt(dbQuery(`SELECT id FROM libri WHERE titolo='${title}' AND deleted_at IS NULL ORDER BY id DESC LIMIT 1`), 10);
-    expect(id).toBeGreaterThan(0);
+    // Poll for the row instead of reading once: submitBook()'s waits are
+    // best-effort (all swallow with .catch), so under parallel workers the
+    // SELECT can race the post-confirm commit and read 0. Retry until the book
+    // is visible.
+    let id = 0;
+    await expect.poll(
+      () => {
+        id = parseInt(dbQuery(`SELECT id FROM libri WHERE titolo='${title}' AND deleted_at IS NULL ORDER BY id DESC LIMIT 1`), 10) || 0;
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/issue-173-cover-save.spec.js]: Fixed in b2ba4e27. The test now passes the UTF-8 title as a MySQL hex literal and compares it with BINARY, so no interpolated SQL string remains. Coverage now uses L’arte, and the create flow explicitly waits for the post-create redirect before opening the edit page.

File: version.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "Pinakes",
-    "version": "0.7.60",
-    "description": "Library Management System - Sistema di Gestione Bibliotecaria"
+  "name": "Pinakes",
+  "version": "0.7.61",
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply version.json]: No code change is appropriate here. Commit 32066591 intentionally promotes the release metadata from the RC line to stable 0.7.61; the PR title and release-policy gates agree. The branch name is historical and is not the released version source.

File: tests/loan-coherence-audit.unit.php [hunk 3]
```diff
@@ -303,6 +309,39 @@
...
+        < strrpos($expiredCron, '$db->commit()');
+$checks['demo seed never authors book state and recalculates before commit'] =
+    !str_contains($demoSeed, "genere_id=?, stato='disponibile'")
+    && str_contains($demoSeed, 'recalculateBookAvailability($bookId, insideTransaction: true)')
+    && strpos($demoSeed, 'recalculateBookAvailability($bookId, insideTransaction: true)')
+        < strpos($demoSeed, '$db->commit()');
+$checks['Book Club acquisition derives availability inside its transaction'] =
+    str_contains($bookClubRepo, 'recalculateBookAvailability($libroId, insideTransaction: true)')
+    && strpos($bookClubRepo, '$this->createCatalogueBookFromExternal(')
+        < strpos($bookClubRepo, '$this->db->commit()', strpos($bookClubRepo, 'public function acquireExternalBook('));
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/loan-coherence-audit.unit.php:340]: Fixed in a0763ea8. The audit now extracts createCatalogueBookFromExternal() and requires the inside-transaction recalculation in that method, while retaining the acquisition-call-before-commit ordering assertion.

File: installer/database/migrations/migrate_0.7.61-rc.1.sql [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    -- domain rather than silently stopping at 1,000 and letting the following
+    -- availability recalculation lower a legacy counter.
+    SELECT th.d * 1000 + h.d * 100 + t.d * 10 + u.d + 1 AS n
+    FROM (SELECT 0 AS d UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) th
+    CROSS JOIN (SELECT 0 AS d UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) h
+    CROSS JOIN (SELECT 0 AS d UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) t
+    CROSS JOIN (SELECT 0 AS d UNION ALL SELECT 1 UNION ALL SELECT 2 UNION ALL SELECT 3 UNION ALL SELECT 4 UNION ALL SELECT 5 UNION ALL SELECT 6 UNION ALL SELECT 7 UNION ALL SELECT 8 UNION ALL SELECT 9) u
+    WHERE th.d * 1000 + h.d * 100 + t.d * 10 + u.d < 9999
+) seq
+    ON seq.n <= GREATEST(legacy.legacy_total - legacy.circulating_rows, 0);
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply installer/database/migrations/migrate_0.7.61-rc.1.sql:101]: Fixed in a0763ea8. Both generated sequence domains are now bounded by the actual maximum remaining deficit, recomputed before each pass and still capped at 9,999. The migration regression test asserts both bounds and passes against the database sandbox.

File: tests/book-field-types.unit.php [hunk 5]
```diff
@@ -194,16 +194,29 @@ function eq($expected, $actual, string $desc): void
...
+
+$totalBefore = $col($id, 'copie_totali');
+$availableBefore = $col($id, 'copie_disponibili');
+$repo->updateBasic($id, [
+    'titolo' => 'ZZ_BFT_derived_guard_' . bin2hex(random_bytes(3)),
+    'stato' => 'disponibile',
+    'copie_totali' => 9999,
+    'copie_disponibili' => 9999,
+]);
+eq('disponibile', $col($id, 'stato'), 'C5 updateBasic ignores a submitted derived stato');
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply tests/book-field-types.unit.php:217]: Fixed in a0763ea8. C5 now submits the conflicting state prestato while asserting that the persisted derived state remains disponibile; the complete 30-case unit passes.

File: installer/database/migrations/migrate_0.7.61-rc.1.sql [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+          AND p5.copia_id IS NULL
+          AND p5.stato IN ('in_corso', 'in_ritardo', 'da_ritirare', 'prenotato')
+    ) needed ON needed.libro_id = c1.libro_id
+    WHERE c1.stato = 'disponibile'
+      AND NOT EXISTS (
+          SELECT 1 FROM `prestiti` p4
+          WHERE p4.copia_id = c1.id
+            AND (p4.attivo = 1 OR (p4.attivo = 0 AND p4.stato = 'pendente'))
+      )
+) cr ON cr.libro_id = lr.libro_id AND cr.rn = lr.rn
```
[the remark this answers is not in the document]
Comment (fabiodalez-dev) [reply installer/database/migrations/migrate_0.7.61-rc.1.sql:235]: Fixed in a0763ea8. README.md still declares MySQL 5.7+, so raising the requirement would be incorrect. Pass C now uses indexed temporary rank tables populated with deterministic correlated counts, without window functions, and keeps the loan-to-copy update inside the same transaction. The 32-case migration test passes.

**Comment by fabiodalez-dev** [comment]
@coderabbitai review

**Comment by fabiodalez-dev** [comment]
@coderabbitai review
